### PR TITLE
ticdc(test): add more ut for kafka sink v2 

### PR DESCRIPTION
**Pull request by charleszheng44**
### What problem does this PR solve?
Issue Number: ref #8241 

### What is changed and how it works?
Add more ut for kafka sink v2, increase the coverage rate of factory.go from 20% to 37.1%

#### Tests 

 - [x] Unit test

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: pkg/sink/kafka/v2/factory_test.go [hunk 1]
```diff
@@ -15,10 +15,14 @@ package v2
 
 import (
 	"context"
+	"crypto/tls"
 	"testing"
 
 	"github.com/pingcap/tiflow/cdc/model"
+	tiflowErr "github.com/pingcap/tiflow/pkg/errors"
```
Comment (3AceShowHand) [inline pkg/sink/kafka/v2/factory_test.go:22]: we always rename it to `cerror`

Comment (charleszheng44) [reply pkg/sink/kafka/v2/factory_test.go]: fixed

File: go.mod [hunk 2]
```diff
@@ -63,6 +63,7 @@ require (
 	github.com/pingcap/tidb v1.1.0-beta.0.20230210031359-532689cb0003
 	github.com/pingcap/tidb-tools v6.5.1-0.20230208065359-62b90e1e24a7+incompatible
 	github.com/pingcap/tidb/parser v0.0.0-20230210031359-532689cb0003
+	github.com/pkg/errors v0.9.1
```
Comment (sdojjy) [inline go.mod:66]: why import this errors package?

Comment (charleszheng44) [reply go.mod]: removed. Use cerror instead

**Review by 3AceShowHand — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

**Pull request merged**